tcms/testcases/static/testcases/js/search.js

```diff
@@ -118,6 +118,37 @@ $(document).ready(function() {
 
     hookIntoPagination('#resultsTable', table);
 
+    const selectAllButton = $("#check-all")
+
+    selectAllButton.click(function () {
+        const rowCheckboxInputButton = $("input:checkbox[name='row-check']")
+        if (selectAllButton.prop("checked")) {
+            rowCheckboxInputButton.prop("checked", true);
+            table.rows().select()
+        } else {
+            rowCheckboxInputButton.prop("checked", false);
+            table.rows().deselect()
+        }
+    });
+
+    table.on('select', function (e, dt, type, indexes) {
+        if (type === 'row') {
+            const totalCheckBoxes = $("input:checkbox[name='row-check']").length;
+            const totalCheckedBoxes = $("input:checkbox[name='row-check']:checked").length;
+            if (totalCheckBoxes === totalCheckedBoxes) {
+                selectAllButton.prop("checked", true)
```

Comment:
instead of the if condition you can pass the boolean expression directly as the second argument.

Reply:
I'm not too sure about that. As we wouldn't want to set "checked" to false if they are not equal.

Reply:
Why not? The only way selectAllButton is checked is when all other rows are checked as well. In every other case it will be unchecked so at the most you will be setting the state of this button to unchecked (which it will be anyway) unconditionally.

This trades an unconditional action (which wont change anything anyway in most of the cases)  for a conditional expression. Both will have the same weight on the JavaScript engine, one of them is slightly less complex.

Reply:
Sorry, I got that mistaken, you're right.